artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueConsumersImpl.java

```diff
@@ -86,11 +86,21 @@ public boolean add(T t) {
       return result;
    }
 
+   @Override
+   public void removed(T t) {
+      iterator.removed(t);
+   }
+
    @Override
    public boolean remove(T t) {
+      iterator.removed(t);
       boolean result = consumers.remove(t);
       if (result) {
          iterator.update(consumers.resettableIterator());
+         if (consumers.isEmpty()) {
+            // TODO: can I always reset?
+            reset();
+         }
```

Comment:
Did you decide no?

Calling the 'external' reset() method instead of iterator.reset() when everything else in the method uses the iterator and consumers objects directly seems a bit odd.

Reply:
I will keep it.. 
and remove it